### ISSUE TITLE
React warning when using `aria-labelledby` prop with `<Checkbox>` component

### ISSUE THREAD

**Issue by breadadams** [tailwindlabs/headlessui#3569]
**What package within Headless UI are you using?**

@headlessui/react

**What version of that package are you using?**

2.2.0

**What browser are you using?**

N/A

**Reproduction URL**

https://codesandbox.io/p/devbox/headlessui-aria-labelledby-warning-y9yjjy

**Describe your issue**

When providing an `aria-labelledby` prop to the `<Checkbox>` component it doesn't work, you have to use `aria-labelledBy` (uppercase `B`), which causes the following warning in the console:

<img width="762" alt="image" src="https://github.com/user-attachments/assets/7cc81f47-5bca-4696-bb37-b77d9e7e237a">

We've also noticed that the `aria-labelledby` isn't autosuggested like other `aria-*` props are, for example `aria-label`.

Can confirm that the same occurs with at least the `<Switch>` component.

**Comment by breadadams** [comment]
This appears to also be happening with other components such as `Input`, `Radio`.

Happy to help here if someone could point me in the right direction, tried finding the source but couldn't track down where the mapping is occurring.

cc @RobinMalfait
